o Make do_prompt(), which has no reason to be global, non-global.
o Don't fail because fread() returns a number different than st.st_size,
  because we asked it to read pointers, not chars.

--- Linux-PAM-0.77/modules/pam_issue/pam_issue.c	2001-11-11 21:23:29.000000000 -0500
+++ Linux-PAM-0.77/modules/pam_issue/pam_issue.c	2003-06-27 17:50:57.000000000 -0400
@@ -38,7 +38,7 @@
 
 static int _user_prompt_set = 0;
 
-char *do_prompt (FILE *);
+static char *do_prompt (FILE *);
 
 /* --- authentication management functions (only) --- */
 
@@ -111,7 +111,7 @@
 		return PAM_IGNORE;
 	    }
 	    memset (prompt_tmp, '\0', st.st_size + 1);
-	    count = fread(prompt_tmp, sizeof(char *), st.st_size, fd);
+	    count = fread(prompt_tmp, sizeof(char), st.st_size, fd);
 	    if (count != st.st_size) {
 		free(prompt_tmp);
 		return PAM_IGNORE;
@@ -164,7 +164,7 @@
      return PAM_IGNORE;
 }
 
-char *do_prompt(FILE *fd)
+static char *do_prompt(FILE *fd)
 {
     int c, size = 1024;
     char *issue = (char *)malloc(size);
